### PR TITLE
Dev/agp/fix warnings

### PR DESCRIPTION
**Pull request by Gnistrup**
Some warnings was quite bad and could cause some undefined behaviour.

### CODE REVIEW

File: STM32/DC/InputValidation/Src/inputValidation.c [hunk 1]
```diff
@@ -37,7 +37,7 @@ int getNumberDigits(int inputNumber){
 bool isInputInt(char * inputBuffer, int startIdx, int endIdx) {
 	int idx = startIdx;
 	while (idx <= endIdx) {
-		if (!isdigit(inputBuffer[idx])) {
+		if (!isdigit((int)inputBuffer[idx])) {
```
Comment (matias-cphatomics) [inline STM32/DC/InputValidation/Src/inputValidation.c:40]: Will this not cause a casting error if it is not a digit? I think it states in the documentation of isdigit() that you have to pass a char.

File: STM32/DC/InputValidation/Src/inputValidation.c [hunk 2]
```diff
@@ -92,7 +92,7 @@ struct actuationInfo parseAndValidateInput(char * inputBuffer){
 	}
 
 	// Check if X in pX is int and valid
-	if (!isdigit(inputBuffer[1]) || atoi(&inputBuffer[1]) > 4 || atoi(&inputBuffer[1]) == 0){
+	if (!isdigit((int)inputBuffer[1]) || atoi(&inputBuffer[1]) > 4 || atoi(&inputBuffer[1]) == 0){
```
Comment (matias-cphatomics) [inline STM32/DC/InputValidation/Src/inputValidation.c:95]: Same as above.

File: STM32/DC/USB_DEVICE/App/usbd_cdc_if.c [hunk 3]
```diff
@@ -247,19 +247,13 @@ static int8_t CDC_Control_FS(uint8_t cmd, uint8_t* pbuf, uint16_t length)
...
 
-    case CDC_SET_CONTROL_LINE_STATE:{
+    case CDC_SET_CONTROL_LINE_STATE: {
         USBD_SetupReqTypedef * req = (USBD_SetupReqTypedef *)pbuf;
-        if(req->wValue &0x0001 != 0){
-      	  isComPortOpen = true;
-        } else {
-          isComPortOpen = false;
-        }
+        isComPortOpen = ((req->wValue & 0x0001) != 0);
```
Comment (matias-cphatomics) [inline STM32/DC/USB_DEVICE/App/usbd_cdc_if.c:252]: Are we sure the only  states  (req->wValue is & 0x0001) can take are 0 and 1? If yes, then fine. If no, then we should check this as the isComPortOpen could be set wrongly.

File: STM32/DC/InputValidation/Src/inputValidation.c (hunk 1 above)
Comment (matias-cphatomics) [reply STM32/DC/InputValidation/Src/inputValidation.c:40]: I read the documentation wrong, my bad :)

**Review by matias-cphatomics — APPROVED** [review]

File: STM32/DC/USB_DEVICE/App/usbd_cdc_if.c (hunk 3 above)
Comment (Gnistrup) [reply STM32/DC/USB_DEVICE/App/usbd_cdc_if.c:252]: Yep I am sure. (req->wValue & 0x0001) is logic bit operator, i.e. only the last bit will remain.

**Pull request merged**